rest/test/plugins/metadata/metadataRoutes_spec.js

```diff
@@ -0,0 +1,466 @@
+						ordering: 'id'
+					}
+				};
+				const route = mockServer.routes['/namespace/:namespaceId/metadata'];
+
+				// Act:
+				return mockServer.callRoute(route, req).then(() => {
+					// Assert:
+					expect(dbGetMetadataWithPaginationFake.calledOnce).to.equal(true);
+					expect(dbGetMetadataWithPaginationFake.firstCall.args[0]).to.equal(metadata.metadataType.namespace);
+					expect(dbGetMetadataWithPaginationFake.firstCall.args[1]).to.deep.equal(accountFilter);
+					expect(dbGetMetadataWithPaginationFake.firstCall.args[2]).to.equal(pagingId);
+					expect(dbGetMetadataWithPaginationFake.firstCall.args[3]).to.equal(10);
+					expect(dbGetMetadataWithPaginationFake.firstCall.args[4]).to.equal(1);
+
+					expect(mockServer.send.firstCall.args[0]).to.deep.equal(processedMetadataOutput);
+					expect(mockServer.next.calledOnce).to.equal(true);
+				});
+			});
+		});
+
+		describe('get metadata by key', () => {
+			it('can get metadata by key', () => {
+				// Arrange:
+				const accountFilter = { 'metadataEntry.targetId': longTypeNamespaceId };
+				const req = { params: { namespaceId, key: scopedMetadataKey } };
+				const route = mockServer.routes['/namespace/:namespaceId/metadata/:key'];
+
+				// Act:
+				return mockServer.callRoute(route, req).then(() => {
+					// Assert:
+					expect(dbGetMetadataByKeyFake.calledOnce).to.equal(true);
+					expect(dbGetMetadataByKeyFake.firstCall.args[0]).to.equal(metadata.metadataType.namespace);
+					expect(dbGetMetadataByKeyFake.firstCall.args[1]).to.deep.equal(accountFilter);
+					expect(dbGetMetadataByKeyFake.firstCall.args[2]).to.deep.equal(uInt64ScopedMetadataKey);
+					expect(mockServer.send.firstCall.args[0]).to.deep.equal(processedMetadataByKeyOutput);
+					expect(mockServer.next.calledOnce).to.equal(true);
+				});
+			});
+		});
+
+		describe('get metadata by key and signer', () => {
+			it('can get metadata by key and signer', () => {
+				// Arrange:
+				const accountFilter = { 'metadataEntry.targetId': longTypeNamespaceId };
+				const req = { params: { namespaceId, key: scopedMetadataKey, publicKey: senderPublicKey } };
+				const route = mockServer.routes['/namespace/:namespaceId/metadata/:key/signer/:publicKey'];
+
+				// Act:
+				return mockServer.callRoute(route, req).then(() => {
+					// Assert:
+					expect(dbGetMetadataByKeyAndSignerFake.calledOnce).to.equal(true);
+					expect(dbGetMetadataByKeyAndSignerFake.firstCall.args[0]).to.equal(metadata.metadataType.namespace);
+					expect(dbGetMetadataByKeyAndSignerFake.firstCall.args[1]).to.deep.equal(accountFilter);
+					expect(dbGetMetadataByKeyAndSignerFake.firstCall.args[2]).to.deep.equal(uInt64ScopedMetadataKey);
+					expect(dbGetMetadataByKeyAndSignerFake.firstCall.args[3]).to.deep.equal(uint8SenderPublicKey);
+					expect(mockServer.send.firstCall.args[0]).to.deep.equal(processedMetadataByKeyAndSignerOutput);
+					expect(mockServer.next.calledOnce).to.equal(true);
+				});
+			});
+		});
+	});
+});
```

Comment:
i think overall, route testing needs more work, we will talk about it, but consider the previous refactor (of `metadataRouteUtils.js`), it might cover the issues i see with them